### PR TITLE
feat/944 - Extension updates - BIP44/Zip32/Ledger/Masp

### LINKED ISSUE #944: Extension: BIP44 Derivation UI
Our SDK package currently supports custom derivation paths for BIP44:

```typescript
const bip44Path = { account: 0, change: 0, index: 1 };
const keys = sdk.keys.deriveFromSeed(seed, bip44Path);
```

There are [designs](https://www.figma.com/design/NFyHbLZXBSl3aUsMxtffvV/Namada-End-User-Interface-V1.0-%2F-Phase-1%3A-Block-Party?node-id=6361-75215&t=vhxKQW05qTF6srmX-1) for specifying a custom path when importing or generating a mnemonic in the Extension `Setup`, as well as from the accounts overview (derive from existing seed)

- Add UI to specify path during `Setup` (`/account'/change'/index'`)
- Add UI to derive keys with custom path against an existing mnemonic (_NOTE_: When a mnemonic account is generated, it can be considered a "parent" account, and other keys can be generated from that mnemonic "seed", which will generate new private/public keys & an address)

Figma:

https://www.figma.com/design/NFyHbLZXBSl3aUsMxtffvV/Namada-End-User-Interface-V1.0-%2F-Phase-1%3A-Block-Party?node-id=6361-75215&t=vhxKQW05qTF6srmX-1

### LINKED ISSUE #1154: Extension: Include version number in Settings
It would be useful to be able to see what version of the extension is running within the extension itself. We originally discussed this a while ago, it just never got implemented.

More info _TBD_

### PR DESCRIPTION
**Pull request by jurevans**
Resolves #944 
Resolves #1154 

- [x] Include Extension/SDK versions in Settings
- [x] Add SDK version to Namadillo settings
- [x] Zip32 should derive default shielded accounts from ed25519 private key (like Namada CLI) with default account
  - [x] Add SDK package functionality to do this from either a Seed (64 bytes) or SK (32 bytes)
    - `sdk.keys.deriveShieldedFromSeed()`
    - `sdk.keys.deriveShieldedFromPrivateKey()` 
- [x] Add BIP44 path form to Mnemonic Import (Advanced)
  - [x] Combine Bip39 passphrase and path to Advanced view
  - [x] Generate shielded account from Zip32 version of path (`makeSaplingPathArray(path)`)
- [x] Add BIP44 path form to Ledger Import (Advanced)
  - _NOTE_ I hooked up shielded keys to our Ledger class, though it won't be useful _until_ we have Zip32 derivation on Ledger. This gives the SDK the interface to use, but we'll want to hook it up to the UI after Zip32 is added!
- [x] Bump extension version
- [x] Hook up Viewing Key
  - [x] Update styles for Viewing Key (See screenshots at bottom)
- [x] General UI Styles (will follow-up after this PR to finish the styles prior to release)

### TODO
- [ ] Add BIP44 path form to Mnemonic Generate (Advanced) - We didn't specify this as a use-case, but it could be added as well - we should discuss that it is also possible to specify a BIP39 passphrase here, though we had copy indicating that it was _only_ to support the same accounts in Namada CLI that used them. If someone used a BIP39 passphrase there, I feel like they could use it on their _generated_ BIP39 seed and not just the imported one.
  - [ ] Generate shielded account from Zip32 version of path (`makeSaplingPathArray(path)`)
- [ ] We can now generate shielded keys from a private key, so this can be added to `Import` -> `Private Key` using a default path
- [ ] Regenerate docs if `sdk` or `types` changes! __NOTE__ This should be in a separate, SDK-release PR, as other PRs will affect the docs too and it creates too much noise for an already large PR

### Screenshots

_NOTE_ Styles are still _in progress_...

Advanced Menu (Import Mnemonic):
Screen Shot 2024-10-18 at 11 53 14 AM

Custom Derivation Path:
Screen Shot 2024-10-18 at 11 53 01 AM

Bip39 Passphrase:
Screen Shot 2024-10-18 at 11 53 22 AM

Keychain & SDK versions:
Screen Shot 2024-10-02 at 1 34 25 PM

SDK version in Namadillo:
Screen Shot 2024-10-03 at 7 39 37 AM

Path displayed (only when a custom path was specified):
Screen Shot 2024-10-31 at 11 41 12 AM

### Designs

Updated designs for `Viewing Key`:
Screenshot 2024-10-09 at 16 29 44

FIGMA:
https://www.figma.com/design/NFyHbLZXBSl3aUsMxtffvV/Namada-End-User-Interface-V1.0-%2F-Phase-1%3A-Block-Party?node-id=6361-75215&node-type=canvas&t=cJGwuzI3Q8soSDus-0

### CODE REVIEW

File: apps/extension/src/Approvals/ConfirmSignLedgerTx.tsx [hunk 1]
```diff
@@ -125,10 +125,12 @@ export const ConfirmSignLedgerTx: React.FC<Props> = ({ details }) => {
...
         }
-        const path = makeBip44Path(
-          chains.namada.bip44.coinType,
-          accountDetails.path
-        );
+        const path = {
+          account: accountDetails.path.account,
+          change: accountDetails.path.change || 0,
+          index: accountDetails.path.index || 0,
+        };
```
Comment (jurevans) [inline apps/extension/src/Approvals/ConfirmSignLedgerTx.tsx:132]: For signing with transparent keys, all of these values are expected to be present!

**Review by mateuszjasiuk — APPROVED** [review]
This works great!

I have couple of minor UI/UX comments.
I personally do not like these numeric input arrows :D 
arrows

Also maybe we can reduce font size or sth so we do not see the scroll.
vk

I'm curious if it makes sense to enable "advanced" button only if we are in Step 2 of HW Wallet import flow. Not sure xd

**Comment by jurevans** [comment]
> 
> I'm curious if it makes sense to enable "advanced" button only if we are in Step 2 of HW Wallet import flow. Not sure xd

Thanks! Yeah some of these style things I will fix in a follow-up PR before release, I think those are all good suggestions!

**Comment by jurevans** [comment]
> I have couple of minor UI/UX comments.
> I personally do not like these numeric input arrows :D

FYI: Tracking these comments in https://github.com/anoma/namada-interface/issues/1229 !

**Pull request merged**